code/campaignclassic/src/main/java/com/adobe/marketing/mobile/InputBoxPushTemplate.java

```diff
@@ -0,0 +1,60 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+package com.adobe.marketing.mobile;
+
+import androidx.annotation.NonNull;
+
+import com.adobe.marketing.mobile.util.DataReader;
+
+import java.util.Map;
+
+public class InputBoxPushTemplate extends AEPPushTemplate {
+    // Optional, Placeholder text for the text input field
+    private String inputFieldText;
```

Comment:
All the member variables can be `final` here in this class as they are to be assigned from the constructor.